src/docs/getting-started/theming.mdx

```diff
@@ -0,0 +1,100 @@
+---
+name: Theming
+menu: 'Getting Started'
+route: /getting-started/theming
+---
+
+# Theming
+
+From the very beginning, React UI has been designed with a great emphasis on
+customizability. We decided to pick CSS custom properties for this feature for
+two main reasons:
+
+1. We believe in **power of native CSS**. Preprocessors are still a thing, but
+   it's not necessary to go as far as for CSS-in-JS to make a UI customizable.
+
+2. Thanks to its JavaScript API, CSS custom properties are both **readable and
+   writable from JS code**.
+
+## Theming Options
+
+CSS custom properties are used to define common visual properties like colors,
+fonts, borders, shadows, or spacing. They come prefixed with `rui-` so they
+don't get in way of other custom properties in your project.
+
+Example `theme.scss`:
+
+```scss
+:root {
+  // Brand colors
+  --rui-color-primary: #00778b;
+  --rui-color-primary-light: #{lighten(#00778b, 70%)};
+  --rui-color-primary-dark: #{darken(#00778b, 4%)};
+  --rui-color-primary-darker: #{darken(#00778b, 8%)};
+  --rui-color-on-primary: #fff;
+  --rui-color-secondary: #fa4616;
+  --rui-color-secondary-light: #{lighten(#fa4616, 45%)};
+  --rui-color-secondary-dark: #{darken(#fa4616, 4%)};
+  --rui-color-secondary-darker: #{darken(#fa4616, 8%)};
+  --rui-color-on-secondary: #fff;
+}
+```
+
+It is also possible to adjust some properties on individual components level,
+preferably by reusing the global settings:
+
+```scss
+:root {
+  // Alerts: common properties
+  --rui-alert-border-width: var(--rui-border-width);
+  --rui-alert-border-radius: var(--rui-border-radius);
+  --rui-alert-padding: var(--rui-spacing-2);
+}
+```
+
+See the [default theme](https://github.com/react-ui-org/react-ui/blob/master/src/lib/theme.scss)
+for the full list of available settings.
+
+️👉 Please note that **breakpoint values are exported as read-only** since CSS
+custom properties
+[cannot be used within media queries](https://www.w3.org/TR/css-variables-1/#using-variables)
+(because media query is not a CSS property).
+
+️👉️ Please note that CSS custom properties are only supported by modern browsers.
+Refer to [caniuse.com](https://caniuse.com/#feat=css-variables) for
+compatibility overview.
+
+## Best Practices
+
+### Customization Approaches
+
+There are two approaches to customization of your project:
+
+1. **Load the default React UI theme** as is and **override** just selected
+   settings in your project's theme. This approach is useful when you are
+   going to make just a couple of small adjustments, and it also performs
+   better.
+
+2. **Copy the default React UI theme** to your project and **change**
+   whatever is needed. The default theme is not loaded in this case. This
+   approach is better for making major visual changes. Just **remember to always
+   check for any new custom properties when upgrading React UI** in your
+   project.
+
+### The Process of Customization
```

Comment:
Well I appreciate _your_ common sense 🙂, but I'm afraid I'm more familiar what less experienced frontend developers are capable of. This is just a friendly guide how it can be done without time-consuming experiments.

Reply:
Also, this is very important:

> Just remember to always check for any new custom properties when upgrading React UI in your project.

Reply:
> Just remember to always check for any new custom properties when upgrading React UI in your project.

True. 
The more I think about it, the more I think this is the most important sentence. In case we add new props what is the correct way it should behave?

1. The build should fail and the developer is forced to add the missing properties.
2. The build passes and uses defaults for new properties.

The problem with 2. is that it can silently change the looks of the application. In case the error is to be visible only in one specific component it can lead to a hard to catch visual glitches. Depending on the developers checking for new properties upon update (would adding properties even be BC?) is in my view fragile. 

Maybe we should only offer two options:
1. Load default file and use the looks of default RUI. This will work as there is no custom design.
2. Copy the default file and use that as base and then load another file with your modifications. This will work as all new properties will break build and force fixing the issue.

Do resolve this at will.

Reply:
> would adding properties even be BC?

That would turn the semver principle upside down 🙂.

I think [this is the answer](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#Custom_property_fallback_values):

```css
.class {
  color: var(--my-var, red); /* Red if --my-var is not defined */
}
```

I discovered this feature not long ago so we'll need to deal with that in the entire library.

Reply:
@adamkudrna what you are proposing is the same as point 2 in my comment above. If the new property default would change the look of a component it should:

* be BC change
* not have a default so the build fails and the developer is forced to declare the new prop explicitly

I'm not sure how big is a chance of a new property changing the visual of existing components, that is up to you to decide.

Reply:
> what you are proposing is the same as point 2 in my comment above

@mbohal Not exactly but I think I know what you mean. The difference is that the default theme doesn't have to be loaded at all. The example code in my previous post could be adapted and read like this:

“Hey RUI `ScrollView` component, try fetching the value of `--my-var` custom property and use it as color of `.class`, and if you don't find it, use red.”

In other words, the fallback si built **right into components** and it makes use of plain CSS for that — the default theme could then be omitted entirely.

As for the breaking change, I'm not sure if I want to be so strict. In general two scenarios can typically happen when new custom properties are added:

1. **A new component is added** and it comes with a set of adjustable custom properties. That would be a feature.
2. **An existing component:**
    1. enables customizing something which wasn't possible before – if it doesn't change its appearance, it's a feature
    2. changes its default appearance – that would undoubtedly be a breaking change

To sum it up, changing the default appearance, customizable or not, means a breaking change. My opinion is we should avoid that unless we _do_ want to release a new, major version of the library — and v1 should slow us down in this.

Reply:
So we can get rid of this section after all!